### DISCUSSION TITLE
Ideas for Yeast, Honey, Water and Adjunct data

### DISCUSSION THREAD

**Discussion by Darkwater23** [Darkwater23/MeadMate#80] [Ideas]
Moved over from issue #76 

**From TravisEvashkevich:**

Love the app, been using for a few meads and recommending around in the communities.

One thing that I'd love to have included when doing a new batch:
Yeast - text? Or drop-down of used yeasts?
Honey varietal - text
Honey weight - number? (Units might make it a pita)
Water amount
Adjuncts (text box like note or something to keep track of fruit or what not)

Make things a bit easier to make sure everything is tracked?

**From Darkwater23:**

Thanks for opening an issue! I enjoy getting feedback and insights into how people make mead and use Mead Mate.

A new yeast feature has merit. I know because I have a My Recipe record that is just all Red Star yeasts and some notes about their alcohol tolerances and flocculation. smile

The varietal field sounds good. You hit the nail on the head with the "units" part on honey weight.

I saw a website for homebrewing while I was researching and learning. That site had the line-item approach where you put in a text label, set a unit, set a quantity, insert a row, and repeat. It seemed very tedious to me. I didn't want that so I just did a big text box.

From a data representation perspective, I also didn't see myself doing anything like, select all batches where water_unit = gallon and amount > 3. After I bottle a batch and maybe create a new recipe entry, I hide and eventually delete the batch record. I personally don't store every batch I ever made in the app. But, that's just me. There's no reporting or export function in the app (yet) so it wasn't necessary to make a more complex data model.

Adding the fields to the batch table isn't hard, but there are some nuances. For example, Javascript and Sqlite use floating-point numbers and integers, but not decimals. So, I might have to store the number as a string. When dealing in pounds, you can have 2.5 lbs or 2lb 8ozs. That could be one field or two fields that I could drive that with a user preference. Form validation would be more complex, but doable.

Honey and water are fundamental parts of mead, so having their own fields in the database and views isn't weird. I just wonder if "juice is worth the squeeze" on that work. For example, maybe some simple markdown parsing of the recipe for honey would work?

Adjuncts are a very interesting idea because they could be part of the initial recipe or they could be added later. So, that could be part of the initial record and a new Event type.

Along with adjuncts, energizers and nutrition might be useful data points, as well.

Can you provide some more details on what is gained by having separate fields for some of these things and maybe some insight into how you use or would like the use the app so I can get a better idea?

Then, I can break this into units of work for future releases.

Thank you!

**From TravisEvashkevich:**

I think for adjuncts just a text field is probably fine really (event like the others, just as a specific thing other than note so it's easier to see when looking at the event list)

Even for the honey amount, I'd probably just sling it in as a text field so it's just an up to the user to fill in (and optional)

I like being able to look back at my meads and what happened and so far this has been super helpful.

On the note of recipes, what could be cool is having a mead + events -> recipe button? Just as a like
Start:
Yeast, honey, water amount and sg

Events:
Stuff that happened

Maybe readings?(less important for reproduction I think)

But I think that would actually also be a handy feature. Easier to share with people as well (screenshot even)

**From Darkwater23:**

That share a batch with events is a great idea! There's more to mead making than the initial ingredients.

When I get a minute at a keyboard, I'll move this to a Discussion and split out the Issues.

**Comment by Darkwater23** [comment]
I believe #13 covers the Yeast feature (or it could with some tweaking).

**Reply by TravisEvashkevich** [reply]
Yeah I think it could pretty well. 
I think obviously make it so it Could just be yeast name filled in (optional)
Not to throw more work/ideas at you but having some stuff that allows for a user to create the yeast data (as in #13 ) outside of recipe or batch would be potentially good? (I guess potentially having a button that was like "create new yeast" or something that would then take you to the yeast info view and then the "ok/done" button would take you back to your batch? (Obvious issue could be serializing the data the user already had as input while going from batch -> yeast info)

**Comment by Darkwater23** [comment]
Adjuncts Feature added as issue #81.

**Comment by Darkwater23** [comment]
Added issue #82 for viewing batch details with events.
